File: leetcode/cpp/LinkedList.cpp
#include<iostream>
using namespace std;
class Node{
public:
    int value;
    Node *next;
    Node(int val){
        this->value = val;
        this->next = nullptr;
    }
};

class LinkedList{
public:
    Node *head;
    Node *tail;
    LinkedList(){
        this->head = nullptr;
        this->tail = nullptr;
    }
    void push(int value){
        Node *N = new Node(value);
        if(this->head == nullptr){
            this->head = N;
            this->tail = N;
            return;
        }
        this->tail->next = N;
        this->tail = N;
    }
    void show(){
        Node *L = this->head;
        while (L!=nullptr)
        {
            cout << L->value << "\n";
            L = L->next;
        }
        
    }
    void RemoveDuplicate(){
        Node* L = this->head;
        Node* toSave;
        while (L!=nullptr)
        {
            if(L->next != nullptr){
                if(L->value == L->next->value){
                    toSave = L->next;
                    L->next = L->next->next;
                    delete toSave;
                }
            }
            L = L->next;
        }
        
    }
};

int main(){

    LinkedList *L = new LinkedList();

    L->push(1);
    L->push(1);
    L->push(2);
    L->push(3);
    L->push(3);
    
    cout << "before clean"<<"\n";
    L->show();
    cout << "AFTER "<<"\n";
    L->RemoveDuplicate();
    L->show();
}

File: leetcode/js/LinkedList.js
class Node{
    constructor(value){
        this.value = value;
        this.next = null;
    }
}
class LinkedList{
    constructor(){
        this.head = null;
    }
    push(value){
        const N = new Node(value);
        let current = this.head;
        if(current == null){
            this.head = N;
            return;
        }
        let beforeNull;
        while(current!==null){
            beforeNull=current;
            current = current.next;
        }
        beforeNull.next = N;
    }
    pop(){
        
    }
    show(){
        let current = this.head;
        while(current!=null){
            console.log(current.value)
            current = current.next;
        }
    }
}


function showRev(start){
    if(start==null)return;
    showRev(start.next);
    console.log(start.value);
}


const L1 = new LinkedList();

L1.push(12);
L1.push(24);
L1.push(55);
L1.show();
console.log("================ Reversed show ==============")
showRev(L1.head);